common/src/main/java/io/druid/math/expr/Parser.java

```diff
@@ -148,30 +148,14 @@ public void visit(Expr expr)
 
   public static Expr.ObjectBinding withMap(final Map<String, ?> bindings)
   {
-    return new Expr.ObjectBinding()
-    {
-      @Override
-      public Number get(String name)
-      {
-        Number number = (Number)bindings.get(name);
-        if (number == null && !bindings.containsKey(name)) {
-          throw new RuntimeException("No binding found for " + name);
-        }
-        return number;
-      }
-    };
+    return bindings::get;
```

Comment:
There is a tradeoff between "ease of use" and the problems from consuming client errors. Maybe it should be a configurable?